lib/pubsub_client/subscriber.rb

```diff
@@ -0,0 +1,30 @@
+# frozen_string_literal: true
+
+require 'google/cloud/pubsub'
+
+module PubsubClient
+  class Subscriber
+    DEFAULT_CONCURRENCY = 8
+
+    # @param subscription [Google::Cloud::PubSub::Subscription]
+    def initialize(subscription)
+      @subscription = subscription
+    end
+
+    # @param concurrency [Integer]
+    # @param auto_ack [Boolean]
+    def subscribe(concurrency, auto_ack, &block)
+      listener = @subscription.listen(threads: { callback: concurrency }) do |received_message|
+        yield received_message.data
```

Comment:
If you only yield the message data, a client down the road will want the entire message container (e.g. for selective ack'ing, or access to the published_at field). You can address this now with one of several options (or wait until there is an actual use case)

1. `yield received_message` -- simple interface, but will require most consumers to extract the `data` field.
1. `yield received_message.data, received_message` -- most consumers will simply grab the first block variable and be all set (`... do |message|`), more advanced consumers can grab just the second block argument (`... do |_, message|`).
1. `yield received_message.data, received_message.published_at, <any other top level fields consumers might want>` -- verbose and effectively self-documenting.

Reply:
I like option 2. It seems like a good compromise between the others.